tests/ui/pattern/usefulness/impl-trait.rs

```diff
@@ -135,7 +135,7 @@ fn nested_empty_opaque(x: Void) -> X {
         let opaque_void = nested_empty_opaque(x);
         let secretely_void = SecretelyVoid(opaque_void);
         match secretely_void {
-            _ => {} //~ ERROR unreachable
+            _ => {}
         }
```

Comment:
I'm confused, how come this "unreachable" is removed but functions like `option_never2` above still compile? Can you try running just this on its own:
```rust
fn option_never(x: Void) -> impl Copy {
    if false {
        match option_never(x) {
            None => {}
        }
    }
    Some(x)
}
```
This should error as non-exhaustive, right?

Reply:
Oh, I know: when there are no explicit patterns like in `match secretely_void { _ => {} }`, exhaustiveness calls `inhabited_predicate` on the whole type. When there are patterns, it calls it on the subtypes *after normalizing*. So in my `option_never` example it sees that `Some` contains `Void`.

In other words, this makes exhaustiveness inconsistent:
```rust
let secretely_void = SecretelyVoid(opaque_void);
match secretely_void {
    SecretelyVoid(_) => {} // WARN: unreachable
}
match secretely_void {} // ERROR: non_exhaustive
```

Reply:
I think the right thing to do is keep the old normalizing behavior for exhaustiveness. This also solves the breaking change issue with `min_exhaustive_patterns`.